op-node/p2p/gossip.go

```diff
@@ -593,21 +592,22 @@ func (p *publisher) SignAndPublishL2Payload(ctx context.Context, envelope *eth.E
 		return fmt.Errorf("failed to sign execution payload with signer: %w", err)
 	}
 	copy(data[:65], sig[:])
-	return p.publishRawSignedPayload(ctx, uint64(envelope.ExecutionPayload.Timestamp), data)
+	return p.publishRawSignedPayload(ctx, envelope.ExecutionPayload.InferVersion(), data)
 }
 
-func (p *publisher) publishRawSignedPayload(ctx context.Context, timestamp uint64, data []byte) error {
+func (p *publisher) publishRawSignedPayload(ctx context.Context, version eth.BlockVersion, data []byte) error {
 	// compress the full message
 	// This also copies the data, freeing up the original buffer to go back into the pool
 	out := snappy.Encode(nil, data)
 
-	if p.cfg.IsIsthmus(timestamp) {
```

Comment:
Need to be careful here since the unmarshalling code may be using the `cfg.Isthmus` method. So this change could break the roundtrip integrity of marshalling / unmarshalling. 

We should add tests to cover this.

Reply:
The block topic itself should control which unmarshall function is called, so this change should actually be fine. 

Getting good test coverage for this is proving a bit complex...